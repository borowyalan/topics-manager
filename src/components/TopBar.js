import React, { Component } from 'react';
import styled from 'styled-components';
import Button from './Button';
import { connect } from 'react-redux';
import { login, logout } from './../actions';
import Text from './Text';
import Colors from '../styles/Colors';

class TopBar extends Component {
    
    onClick = () => {
        const { loggedUser, login, logout } = this.props;
        const onPressAction = loggedUser ? logout : login;
        onPressAction();
    };
    
    render() {
        const { loggedUser } = this.props;
        const buttonText = loggedUser ? 'Wyloguj' : 'Zaloguj';
        return (
            <TopBarWrapper>
                <Text type="basic" color={ Colors.white } display={ 'inline' }>
                    Zaloguj siÄ™ przez GitHub
                </Text>
                <Button marginLeft='10px' onClick={this.onClick}
                        type="primary">
                    { buttonText }
                </Button>
            </TopBarWrapper>
        );
    }
    
}

const mapStateToProps = state => ({
    loggedUser: state.loggedUser
});

const mapDispatchToProps = dispatch => ({
    login: () => dispatch(login()),
    logout: () => dispatch(logout())
});

const TopBarWrapper = styled.section`
    width: 100%;
    background-color: ${Colors.black};
    color: ${Colors.white};
    padding: 7px;
`;

export default connect(mapStateToProps, mapDispatchToProps)(TopBar);
