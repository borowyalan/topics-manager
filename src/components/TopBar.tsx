import React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
import { requestLogin as loginAction, requestLogout as logoutAction } from '../actions/auth.actions';
import ActivityIndicator from '../shared/components/ActivityIndicator';
import { BackgroundImage } from '../shared/components/BackgroundImage';
import { User } from '../shared/models/user';
import Colors from '../styles/Colors';
import Button, { ButtonType } from './Button';
import Section from './Section';
import Text, { TextType } from './Text';

export interface TopBarProps {
    logged: boolean,
    pending: boolean,
    error?: any,
    user: User,
    requestLogin: () => undefined,
    requestLogout: () => undefined,
}

const ButtonPosition = styled.div`
    margin-left: 15px
`;

class TopBar extends React.Component<TopBarProps> {
    public onClick = () => {
        const { logged, requestLogout, requestLogin } = this.props;
        const onPressAction = logged ? requestLogout : requestLogin;
        onPressAction();
    };

    public render() {
        const { logged, pending, user, error } = this.props;
        const buttonText = logged ? 'Wyloguj' : 'Zaloguj';
        return (
            <Section backgroundColor={Colors.black}>
                <SpaceBetweenContainer>
                    <BackgroundImage url={'http://'}/>
                    <Text kind={TextType.Default}/>
                    <Text kind={TextType.Logo} style={{margin: "0 20px 0 0"}}>
                        Warsaw<strong>JS</strong>
                    </Text>
                    <MenuList>
                        <MenuLink>Meetup</MenuLink>
                        <MenuLink>Worhshop</MenuLink>
                        <MenuLink>Praca</MenuLink>
                        <MenuLink>O nas</MenuLink>
                        <MenuLink>Archiwum</MenuLink>
                        <MenuLink>Formularz</MenuLink>
                        <MenuLink>Kontakt</MenuLink>
                        <MenuLink>Blog</MenuLink>
                    </MenuList>
                    <Text kind={TextType.Basic} color={Colors.white}>
                        {logged && user && user.email}
                    </Text>
                    {pending && <ActivityIndicator/>}
                    {error && <p>Błąd: {JSON.stringify(error)}</p>}

                    <ButtonPosition>
                        <Button
                            onClick={this.onClick}
                            kind={ButtonType.Primary}>
                            {buttonText}
                        </Button>
                    </ButtonPosition>
                </SpaceBetweenContainer>
            </Section>
        );
    }
}

const MenuLink = styled.li`
    color: white;
    &:hover {
        color: ${Colors.red};
        padding: 5px 0;
        cursor: pointer;
        border-bottom: solid 1px ${Colors.white};
    }
`;
const SpaceBetweenContainer = styled.div`
    width: 100%;
    display: flex;
    justify-content: space-between;
    align-items: center;
`;
const MenuList = styled.ul`
    width: 100%;
    display: flex;
    justify-content: space-between;
    align-items: center;
    list-style-type: none;
    margin: 0 10px;
`;

const mapStateToProps = state => ({
    logged: state.auth.logged,
    pending: state.auth.pending,
    user: state.auth.user,
    error: state.auth.error,
});

const mapDispatchToProps = dispatch => ({
    requestLogin: () => dispatch(loginAction()),
    requestLogout: () => dispatch(logoutAction()),
});

export default connect(
    mapStateToProps,
    mapDispatchToProps,
)(TopBar);
