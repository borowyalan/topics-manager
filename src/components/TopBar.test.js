import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureMockStore from 'redux-mock-store';
import 'jest-styled-components';
import TopBar from '../components/TopBar';

configure({adapter: new Adapter() });

const mockStore = configureMockStore();

describe('<TopBar> should render', () => {
    const state = {
        loggedUser: true
    }
    const store = mockStore(state);
    const wrapper = shallow(<TopBar store={store}/>).dive();
    it('should contain certain texts', () => {
        const styledComponent = wrapper.find('Text');
        expect(styledComponent.children().text()).toBe('Zaloguj siÄ™ przez GitHub');
    })

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })
});