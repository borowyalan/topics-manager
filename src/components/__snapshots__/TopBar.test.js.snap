// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopBar> should render should match snapshot 1`] = `
<styled.section>
  <Text
    color="#fff"
    display="inline"
    type="basic"
  >
    Zaloguj siÄ™ przez GitHub
  </Text>
  <Button
    marginLeft="10px"
    onClick={[Function]}
    type="primary"
  >
    Wyloguj
  </Button>
</styled.section>
`;
